File: exercism/clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn is-divisible-by-3 [number]
  (= 0 (rem number 3)))

(defn is-divisible-by-5 [number]
  (= 0 (rem number 5)))

(defn is-divisible-by-7 [number]
  (= 0 (rem number 7)))

(defn convert [number]
  (str (if (is-divisible-by-3 number) "Pling" "")
       (if (is-divisible-by-5 number) "Plang" "")
       (if (is-divisible-by-7 number) "Plong" "")
       (if (and (not (is-divisible-by-3 number)) (not (is-divisible-by-5 number)) (not (is-divisible-by-7 number))) (str number) "")))

File: exercism/python/raindrops/raindrops.py
TEXT_BY_FACTORS = {
    3 : "Pling",
    5 : "Plang",
    7 : "Plong"
}

def convert(number):
    result = [v for k,v in sorted(TEXT_BY_FACTORS.items()) if number % k == 0]
    return "".join(result) if result else str(number)